File: app/src/main/java/com/example/pepper_test2/BasicEmotion.java
package com.example.pepper_test2;

/**
 * Represents a basic emotion derived from excitement and pleasure states.
 * Based on James Russell's emotion model as implemented in QiSDK.
 */
public enum BasicEmotion {
    UNKNOWN("❓", "Unknown"),
    NEUTRAL("😐", "Neutral"),
    CONTENT("😊", "Content"),
    JOYFUL("😄", "Joyful"),
    SAD("😢", "Sad"),
    ANGRY("😠", "Angry");

    private final String emoji;
    private final String displayName;

    BasicEmotion(String emoji, String displayName) {
        this.emoji = emoji;
        this.displayName = displayName;
    }

    /**
     * Get formatted string for dashboard display
     */
    public String getFormattedDisplay() {
        return emoji + " " + displayName;
    }
}

File: app/src/main/java/ch/fhnw/pepper_realtime/data/BasicEmotion.kt
package ch.fhnw.pepper_realtime.data

/**
 * Represents a basic emotion derived from excitement and pleasure states.
 * Based on James Russell's emotion model as implemented in QiSDK.
 */
enum class BasicEmotion(private val emoji: String, private val displayName: String) {
    UNKNOWN("❓", "Unknown"),
    NEUTRAL("😐", "Neutral"),
    CONTENT("😊", "Content"),
    JOYFUL("😄", "Joyful"),
    SAD("😢", "Sad"),
    ANGRY("😠", "Angry");

    /**
     * Get formatted string for dashboard display
     */
    fun getFormattedDisplay(): String {
        return "$emoji $displayName"
    }
}

